packages/metamask/src/index.ts

```diff
@@ -133,10 +145,10 @@ export class MetaMask extends Connector {
       .then(async () => {
         if (!this.provider) throw new NoMetaMaskError()
 
-        return Promise.all([
-          this.provider.request({ method: 'eth_chainId' }) as Promise<string>,
-          this.provider.request({ method: 'eth_requestAccounts' }) as Promise<string[]>,
```

Comment:
haven't tested but i think this is a problem - we can't remove `eth_requestAccounts`, otherwise an unconnected user will never be able to trigger a connection popup